x-pack/plugin/src/yamlRestTest/resources/rest-api-spec/test/esql/161_union_types_nested.yml

```diff
@@ -0,0 +1,203 @@
+setup:
```

Comment:
The word "nested" means the `nested` type to me. Could you rename this file to something like "sub_fields" or something? I forgot what we call those sub field things, but `nested` and `object` mean something specific to me.